specification/2.0/README.md

```diff
@@ -178,6 +178,10 @@ To simplify client-side implementation, glTF has additional restrictions on JSON
 
    > **Implementation Note:** This allows generic glTF client implementations to not have full Unicode support. Application-specific strings (e.g., values of `"name"` properties or content of `extras` fields) may use any symbols.
 3. Names (keys) within JSON objects must be unique, i.e., duplicate keys aren't allowed.
+4. Numbers defined as integers in the schema must be written without a fractional part (e.g., `.0`). Exporters should not produce integer values greater than 2<sup>53</sup> because some client implementations will not be able to read them correctly.
```

Comment:
For the "without a fractional part" requirement, could you include the motivation for that? A value of `1.0` would pass JSON validation for `{ "type": "number", "multipleOf": 1.0 }`, so this requirement is really because some (many?) JSON libraries for typed languages depend on it?

Reply:
@donmccurdy Related: https://github.com/KhronosGroup/glTF-Validator/issues/109

Reply:
Yes, but I think the rationale should be included explicitly in the specification – otherwise it sounds like we're inventing our own subset of JSON, which would be a bad look. 😄

Reply:
Some of the issues that are related to the original `integer` type have already been linked to in other comments. Some of the related hiccups can probably be explained by the fact that earlier versions of the ("official") JSON-Schema included the `integer` type, but it has been removed in the meantime (cf. https://github.com/json-schema-org/json-schema-spec/issues/146 ). 

One specific issue is caused by the combination of a **typed** language and automatic code generation from the schema. In doubt, the code generation has to be tweaked and configured so that a `{ "type": "number", "multipleOf": 1.0 }` is translated into a `long`, and that the resulting parser does a `parseLong` instead of a `parseDouble` ...

Reply:
`"type": "integer"` was added back at some point. Latest definitions are:
- > concept of "integer" is from a vocabulary, not the data model
- > For consistency, integer JSON numbers SHOULD NOT be encoded with a fractional part.
- > String values MUST be one of the six primitive types ("null", "boolean", "object", "array", "number", or "string"), or "integer" which matches any number with a zero fractional part.

Sources:
- https://json-schema.org/draft/2019-09/json-schema-core.html
- https://json-schema.org/draft/2019-09/json-schema-validation.html

Reply:
It seems like there's an [important distinction](https://tools.ietf.org/rfc/rfc2119.txt) between "SHOULD NOT" (the JSON spec wording) and "must be written without a fractional part" (this PR).  The JSON wording doesn't outright prevent encodings like `1.1e1` for integers, although they are strongly discouraged.

It may still make sense for glTF to have stronger enforcement, for the sake of strongly-typed glTF readers that don't wish to take extra steps of parsing floats and rounding.  But I agree with Don's concern that this needs to avoid becoming a subset of JSON.

Reply:
OK, they update the JSON schema draft regularly, and I wonder about their roadmap for a ratification - without that, https://github.com/KhronosGroup/glTF/issues/929 remains a moving target... :-/

Reply:
I think we should copy the JSON wording here, changing "must be written without a fractional part" to "should not be written without a fractional part."